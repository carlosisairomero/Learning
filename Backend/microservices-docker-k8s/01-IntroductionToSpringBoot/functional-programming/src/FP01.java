import java.util.List;

public class FP01 {
    public static void main(String[] args) {
        System.out.println("Structured:");
        printAllNumbersInListStructured(List.of(12, 9, 13, 4, 6, 2, 4, 12, 15));
        System.out.println("\nFunctional:");
        printAllNumbersInListFunctional(List.of(12, 9, 13, 4, 6, 2, 4, 12, 15));
    }

    private static void printAllNumbersInListStructured(List<Integer> numbers) {
        // How to loop the numbers?
        for (int number : numbers) {
            System.out.print(number + " ");
        }
    }

    private static void printAllNumbersInListFunctional(List<Integer> numbers) {
        // What to do?
        numbers.stream()
                .forEach(System.out::print);
    }
}



