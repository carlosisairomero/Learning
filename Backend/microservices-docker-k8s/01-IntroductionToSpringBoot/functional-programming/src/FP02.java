import java.util.List;

public class FP02 {
    public static void main(String[] args) {
        System.out.println("Structured:");
        printEvenNumbersInListStructured(List.of(12, 9, 13, 4, 6, 2, 4, 12, 15));
        System.out.println("\nFunctional:");
        printEvenNumbersInListFunctional(List.of(12, 9, 13, 4, 6, 2, 4, 12, 15));
    }

    private static void printEvenNumbersInListStructured(List<Integer> numbers) {
        // How to do?
        for (int number : numbers) {
            if(number % 2 == 0) {
                System.out.println(number );
            }
        }
    }

    private static void printEvenNumbersInListFunctional(List<Integer> numbers) {
        // What to do?
        numbers.stream()
                .filter(n -> n % 2 == 0)
                .forEach(System.out::println);
    }
}






