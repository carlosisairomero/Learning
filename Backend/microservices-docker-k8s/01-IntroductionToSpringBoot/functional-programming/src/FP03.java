import java.util.List;

public class FP03 {
    public static void main(String[] args) {
        List<Integer> numbers = List.of(12, 9, 13, 4, 6, 2, 4, 12, 15);
        List<String> courses = List.of("Spring", "Spring Boot", "API", "Microservices", "AWS", "PCF", "Azure", "Docker", "Kubernetes");

        System.out.println("E1: Print Only Odd Numbers");
        printOddNumbers(numbers);
        System.out.println("E2: Print Courses containing the word Spring");
        printCoursesThatContainsWordSpring(courses);
        System.out.println("E3: Print Courses with at least 4 letters");
        printCoursesWhoseNameHasAtLeast4Letters(courses);
    }

    private static void printOddNumbers(List<Integer> numbers) {
        numbers.stream()
                .filter(n -> n % 2 != 0)
                .forEach(System.out::println);
    }

    private static void printCoursesThatContainsWordSpring(List<String> courses) {
        courses.stream()
                .filter(c -> c.contains("Spring"))
                .forEach(System.out::println);
    }

    private static void printCoursesWhoseNameHasAtLeast4Letters(List<String> courses) {
        courses.stream()
                .filter(c -> c.length() >= 4)
                .forEach(System.out::println);
    }
}






