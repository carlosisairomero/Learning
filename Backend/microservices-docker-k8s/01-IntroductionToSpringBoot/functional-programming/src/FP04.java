import java.util.List;

public class FP04 {
    public static void main(String[] args) {
        List<Integer> numbers = List.of(12, 9, 13, 4, 6, 2, 4, 12, 15);
        List<String> courses = List.of("Spring", "Spring Boot", "API", "Microservices", "AWS", "PCF", "Azure", "Docker", "Kubernetes");

        System.out.println("E1: Print Squares of Even Numbers");
        printSquaresOfEvenNumbers(numbers);
        System.out.println("E2: Print Cubes of Odd Numbers");
        printCubesOfOddNumbers(numbers);
        System.out.println("E3: Print Number of Chars For Each Course");
        printNumberOfCharactersForEachCourse(courses);
    }

    private static void printSquaresOfEvenNumbers(List<Integer> numbers) {
        numbers.stream()
                .filter(n -> n % 2 == 0)
                .map(n -> n * n)
                .forEach(System.out::println);
    }

    private static void printCubesOfOddNumbers(List<Integer> numbers) {
        numbers.stream()
                .filter(n -> n % 2 != 0)
                .map(n -> n * n * n)
                .forEach(System.out::println);
    }

    private static void printNumberOfCharactersForEachCourse(List<String> courses) {
        courses.stream()
                .map(c -> c + " " + c.length())
                .forEach(System.out::println);
    }

}






