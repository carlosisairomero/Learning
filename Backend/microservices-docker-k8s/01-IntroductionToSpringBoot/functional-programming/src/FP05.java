import java.util.List;
import java.util.Optional;
import java.util.function.Predicate;

public class FP05 {
    public static void main(String[] args) {
        List<String> fruits = List.of("apple", "banana", "mango");
        // I want the first fruit that starts with a "b"
        Predicate<? super String> predicate = f -> f.startsWith("c");
        Optional<String> firstWithB = fruits.stream().filter(predicate).findFirst();
        System.out.println(firstWithB);
        System.out.println(firstWithB.isEmpty());
        System.out.println(firstWithB.isPresent());
        System.out.println(firstWithB.get());
    }
}






