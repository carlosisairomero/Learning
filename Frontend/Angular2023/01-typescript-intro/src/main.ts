import './style.css'
// import './topics/01-basic-types.ts'
// import './topics/02-object-interface.ts'
// import './topics/03-functions.ts'
// import './topics/04-homework-types.ts'
// import './topics/05-basic-destructuring.ts'
// import './topics/06-function-destructuring.ts'
// import './topics/07-import-export.ts'
// import './topics/08-clases';
// import './topics/08.1-clases-composition';
// import './topics/09-generics';
// import './topics/10-decorators';
import './topics/11-optional-chaining';

document.querySelector<HTMLDivElement>('#app')!.innerHTML = `
  Hi
`

