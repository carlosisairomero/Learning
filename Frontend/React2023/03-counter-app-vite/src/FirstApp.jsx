import PropTypes from 'prop-types';

export const FirstApp = ({title, subtitle, name}) => {
   return (
      <>
         <h1 data-testid='test-title'>{title}</h1>
         <div>{subtitle}</div>
         <div>{subtitle}</div>
         <div>{name}</div>
      </>
   )
}

FirstApp.propTypes = {
   title: PropTypes.string.isRequired,
   subtitle: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
   subtitle: 'subtitle',
   name: 'name',
 };


