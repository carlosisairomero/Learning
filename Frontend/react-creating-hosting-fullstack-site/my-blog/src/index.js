import React from 'react';
import ReactDOM from 'react-dom/client';
import './index.css';
import App from './App';
import reportWebVitals from './reportWebVitals';

// Import the functions you need from the SDKs you need
import { initializeApp } from "firebase/app";
// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
const firebaseConfig = {
  apiKey: "AIzaSyAi7M0JVJbU25GAzO_13j1cPPAI8DDeFyQ",
  authDomain: "my-react-blog-fbbb7.firebaseapp.com",
  projectId: "my-react-blog-fbbb7",
  storageBucket: "my-react-blog-fbbb7.appspot.com",
  messagingSenderId: "401841206366",
  appId: "1:401841206366:web:84206735d8d6f8085830cf"
};

// Initialize Firebase
const app = initializeApp(firebaseConfig);

const root = ReactDOM.createRoot(document.getElementById('root'));
root.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
);

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals();
