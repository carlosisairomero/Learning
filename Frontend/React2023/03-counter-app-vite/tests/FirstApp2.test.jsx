import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Test: FirstApp', () => {
   const title = 'Hola, Soy Goku';
   const subtitle = 'Soy un subtitulo';

   test('should match with snapshot', () => {
      const {container} = render(
         <FirstApp title={title}/>
      );
      expect(container).toMatchSnapshot();
   });

   test('should show msg: Hola, Soy Goku', () => {
      render(
         <FirstApp title={title}/>
      );
      expect(screen.getByText(title)).toBeTruthy();
   });

   test('should show msg: Hola, Soy Goku in a h1', () => {
      render(
         <FirstApp title={title}/>
      );
      expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);
   });

   test('should show the subtitle sent by props', () => {
      render(
         <FirstApp 
            title={title} 
            subtitle={subtitle}
         />
      );
      expect(screen.getAllByText(subtitle).length).toBe(2);
   });
});