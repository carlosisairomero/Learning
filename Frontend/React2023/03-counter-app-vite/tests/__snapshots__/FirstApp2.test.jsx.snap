// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test: FirstApp should match with snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, Soy Goku
  </h1>
  <div>
    subtitle
  </div>
  <div>
    subtitle
  </div>
  <div>
    name
  </div>
</div>
`;
